cpp/src/arrow/flight/CMakeLists.txt

```diff
@@ -197,7 +197,9 @@ add_arrow_lib(arrow_flight
               ${ARROW_FLIGHT_LINK_LIBS}
               STATIC_LINK_LIBS
               arrow_static
-              ${ARROW_FLIGHT_LINK_LIBS})
+              ${ARROW_FLIGHT_LINK_LIBS}
+              ${ARROW_STATIC_INSTALL_INTERFACE_LIBS_REAL}
+              arrow_static)
```

Comment:
Hmm, do we need to list arrow_static twice? And it seems odd that we have to re-list all the libraries here…why doesn't listing arrow_static as a dependency give us these dependencies transitively?

Reply:
Indeed. Removed.